### PR TITLE
MOSIP-28549 changes in event-type, authentication-mode and description for multilingual support

### PR DESCRIPTION
**Pull request by RitikJain4108**

### CODE REVIEW

File: resident/resident-service/src/main/java/io/mosip/resident/util/TemplateUtil.java [hunk 1]
```diff
@@ -201,19 +197,32 @@ private String getAttributesDisplayText(String attributesFromDB, String language
...
 
 	public String getDescriptionTemplateVariablesForAuthenticationRequest(
 			ResidentTransactionEntity residentTransactionEntity, String fileText, String languageCode) {
-		return residentTransactionEntity.getStatusComment();
+		List<String> authTypeCodeTemplateValues = new ArrayList<>();
+		String statusCode = residentService.getEventStatusCode(residentTransactionEntity.getStatusCode(), languageCode)
+				.getT1();
+		String authTypeCodeFromDB = residentTransactionEntity.getAuthTypeCode();
+		if (authTypeCodeFromDB != null && !authTypeCodeFromDB.isEmpty()) {
+			List<String> authTypeCodeListFromDB = List
```
Comment (loganathan-sekaran) [inline resident/resident-service/src/main/java/io/mosip/resident/util/TemplateUtil.java]: Use Stream API to map it to the new list, avoid creating separate list and adding elements in for loop.

File: resident/resident-service/src/main/java/io/mosip/resident/util/TemplateUtil.java [hunk 2]
```diff
@@ -123,13 +118,14 @@ public Map<String, String> getCommonTemplateVariables(ResidentTransactionEntity
 		templateVariables.put(TemplateVariablesConstants.PURPOSE, residentTransactionEntity.getPurpose());
 		templateVariables.put(TemplateVariablesConstants.ATTRIBUTE_LIST, getAttributesDisplayText(
 				replaceNullWithEmptyString(residentTransactionEntity.getAttributeList()), languageCode, requestType));
-		String authenticationMode = getAuthTypeCodeTemplateValue(
-				replaceNullWithEmptyString(residentTransactionEntity.getAuthTypeCode()), languageCode);
-		if (authenticationMode.equalsIgnoreCase(UNKNOWN)) {
-			templateVariables.put(TemplateVariablesConstants.AUTHENTICATION_MODE,
-					replaceNullWithEmptyString(residentTransactionEntity.getAuthTypeCode()));
+		String authTypeCode = replaceNullWithEmptyString(residentTransactionEntity.getAuthTypeCode());
+		AuthenticationModeEnum authenticationMode = AuthenticationModeEnum.getAuthenticationModeName(authTypeCode, env);
```
Comment (loganathan-sekaran) [inline resident/resident-service/src/main/java/io/mosip/resident/util/TemplateUtil.java]: Whether multiple comma separated Authentication Modes in the authTypeCode is handled?

**Review by loganathan-sekaran — CHANGES_REQUESTED** [review]

File: resident/resident-service/src/main/java/io/mosip/resident/util/TemplateUtil.java (hunk 1 above)
Comment (RitikJain4108) [reply resident/resident-service/src/main/java/io/mosip/resident/util/TemplateUtil.java]: done

File: resident/resident-service/src/main/java/io/mosip/resident/util/TemplateUtil.java (hunk 2 above)
Comment (RitikJain4108) [reply resident/resident-service/src/main/java/io/mosip/resident/util/TemplateUtil.java]: I am working on it...

Comment (RitikJain4108) [reply resident/resident-service/src/main/java/io/mosip/resident/util/TemplateUtil.java]: done

**Review by loganathan-sekaran — APPROVED** [review]

**Review by loganathan-sekaran — APPROVED** [review]

**Pull request merged**